tasks/main.yml

```diff
@@ -8,15 +8,15 @@
   tags: install-deps
 
 - name: configure web proxy
-  when: azure_pipelines_agent_ctl_proxy["url"] is defined
+  when: azure_pipelines_agent_ctl_proxy.url is defined
   block:
     - name: write {{ azure_pipelines_agent_ctl_install_dir }}/.proxy
       template:
         src: .proxy.j2
         dest: '{{ azure_pipelines_agent_ctl_install_dir }}/.proxy'
         mode: 0644
     - name: write {{ azure_pipelines_agent_ctl_install_dir }}/.proxybypass
-      when: azure_pipelines_agent_ctl_proxy["bypass"] is defined
+      when: azure_pipelines_agent_ctl_proxy.bypass is defined
```

Comment:
I think this needs to be
```yml
when: 'bypass' in azure_pipelines_agent_ctl
```

Reply:
That requires multiple quotes (', ") to be utilized in order to get Ansible to accept the expression; therefore, going forward with the current format (ref: https://serverfault.com/questions/857973/is-there-a-way-to-check-that-a-dictionary-key-is-not-defined-in-ansible-task)